src/Build/Evaluation/ProjectRootElementCache.cs

```diff
@@ -71,22 +71,21 @@ internal class ProjectRootElementCache : ProjectRootElementCacheBase
         /// If this number is increased much higher, the datastructure may
         /// need to be changed from a linked list, since it's currently O(n).
         /// </remarks>
-#pragma warning disable CA1802 // Use literals where appropriate
-        private static readonly int s_maximumStrongCacheSize = 200;
-#pragma warning restore CA1802 // Use literals where appropriate
+        private static readonly int s_maximumStrongCacheSize =
+            Convert.ToInt32(Environment.GetEnvironmentVariable("MSBUILDPROJECTROOTELEMENTCACHESIZE") ?? "200", NumberFormatInfo.InvariantInfo);
```

Comment:
Can you please add a new private static method and:

```suggestion
        private static readonly int s_maximumStrongCacheSize = InitializeMaximumStrongCacheSize();
```

The initialization expression you wrote makes us parse the string `"200"` which is suboptimal.

Reply:
That works too!

Reply:
Done.